src/mrtimer.rs

```diff
@@ -0,0 +1,146 @@
+//! API for the MRT peripheral
+//!
+//! Please be aware that this doesn't try to abstract everything, it only
+//! implements the embedded-hal `Timer` functionality.
+//!
+//! The MRT consists of 4 channels, which are mostly separate and can each act
+//! as a run-of-the-mill timer.
+//!
+//! # Example
+//!
+//! ``` no_run
+//! use lpc8xx_hal::prelude::*;
+//! use lpc8xx_hal::Peripherals;
+//!
+//! use nb::block;
+//!
+//! let mut p = Peripherals::take().unwrap();
+//!
+//! let mut syscon = p.SYSCON.split();
+//! let [mut timer, _, _, _] = p.SYST.enable_delay();
+//! timer.start(12_000_000u32);
+//! loop {
+//!     block!(timer.wait()).unwrap();
+//! }
+//! ```
+
+use crate::{
+    pac::{mrt0::CHANNEL, MRT0},
+    reg_proxy::RegProxy,
+    syscon,
+};
+
+use embedded_hal::timer::{CountDown, Periodic};
+use nb::{Error, Result};
+use void::Void;
+
+/// Represent a MRT0 instance
+pub struct MRTimer {
+    mrt: MRT0,
+}
+
+/// Represent a MRT0 channel
+pub struct MRTimerChannel {
+    channel: u8,
+    channels: RegProxy<CHANNEL>,
+}
+
+impl MRTimer {
+    /// Assumes peripheral is in reset state
+    ///
+    /// This means:
+    /// - Each channel is in repeat mode
+    /// - All channel interrupts are disabled
+    pub(crate) fn new(mrt: MRT0) -> Self {
+        Self { mrt }
+    }
+
+    /// Enables the MRTimer and splits it into it's four channels
+    pub fn split(self, syscon: &mut syscon::Handle) -> [MRTimerChannel; 4] {
+        syscon.enable_clock(&self.mrt);
+        [
+            MRTimerChannel {
+                channel: 0,
+                channels: RegProxy::new(),
+            },
+            MRTimerChannel {
+                channel: 1,
+                channels: RegProxy::new(),
+            },
+            MRTimerChannel {
+                channel: 2,
+                channels: RegProxy::new(),
+            },
+            MRTimerChannel {
+                channel: 3,
+                channels: RegProxy::new(),
+            },
+        ]
+    }
+
+    /// Return the raw peripheral
+    ///
+    /// This method serves as an escape hatch from the HAL API. It returns the
+    /// raw peripheral, allowing you to do whatever you want with it, without
+    /// limitations imposed by the API.
+    ///
+    /// If you are using this method because a feature you need is missing from
+    /// the HAL API, please [open an issue] or, if an issue for your feature
+    /// request already exists, comment on the existing issue, so we can
+    /// prioritize it accordingly.
+    ///
+    /// [open an issue]: https://github.com/lpc-rs/lpc8xx-hal/issues
+    pub fn free(self) -> MRT0 {
+        self.mrt
+    }
+}
+
+// TODO the lpc82x um isn't quite clear on how many bits the mrt has
```

Comment:
The MRT has 31 bits. This has been wrong in the SVD, but has since been fixed (https://github.com/lpc-rs/lpc-pac/issues/18). The comment in the user manual is probably a leftover from whatever device they copy-pasted from.

Reply:
Thanks for the confirmation